### PR TITLE
Add async variants of `XPCClient.send/sendMessage(...)`

### LINKED ISSUE #19: Add async variants of `XPCClient.send/sendMessage(...)` functions
macOS 10.15 and later should support this (at least as of the latest Xcode beta)

### PR DESCRIPTION
**Pull request by jakaplan**
Fixes #19

Adds new tests for the async variants. Updates documentation to match.

### CODE REVIEW

**Comment by jakaplan** [comment]
@amomchilov for your review

File: Sources/SecureXPC/Client/XPCClient.swift [hunk 1]
```diff
@@ -191,6 +219,19 @@ public class XPCClient {
         }
     }
     
+    /// Sends with no message and does not receive a reply.
+    ///
+    /// - Parameters:
+    ///   - route: The server route which will handle this.
+    @available(macOS 10.15.0, *)
+    public func send(route: XPCRouteWithoutMessageWithoutReply) async throws {
+        try await withCheckedThrowingContinuation { continuation in
```
Comment (amomchilov) [inline Sources/SecureXPC/Client/XPCClient.swift:228]: Why `checked`? This looks safe for `withUnsafeThrowingContinuation`

File: Tests/SecureXPCTests/Client & Server/Round-trip Integration Test.swift [hunk 2]
```diff
@@ -20,7 +20,7 @@ class RoundTripIntegrationTest: XCTestCase {
         anonymousServer.start()
     }
 
-    func testSendWithMessageWithReply() throws {
+    func testSendWithMessageWithReply_Sync() throws {
```
Comment (amomchilov) [inline Tests/SecureXPCTests/Client & Server/Round-trip Integration Test.swift:23]: I think it's reasonable to just completely replace these closure-based tests.

In my own work, I've pretty much completely abandoned the XCTest continuation APIs in favour of async/await, even for old projects that don't use it in the production code (I just add wrappers in my test bundles, which makes test writing easier and more correct).

**Review by amomchilov — APPROVED** [review]

File: Tests/SecureXPCTests/Client & Server/Round-trip Integration Test.swift (hunk 2 above)
Comment (jakaplan) [reply Tests/SecureXPCTests/Client & Server/Round-trip Integration Test.swift:23]: I'm going to leave them for now as the behavior differs for two of them depending on whether they have a closure provided or not, but point taken.

**Pull request merged**